lib/src/equality.dart

```diff
@@ -30,6 +30,38 @@ abstract class Equality<E> {
   bool isValidKey(Object o);
 }
 
+typedef F _GetKey<E, F>(E object);
+
+/// Equality of objects using a canonical value of the object itself.
+///
+/// For example, given the class:
+///     abstract class Employee {
+///       int get employmentId;
+///     }
+///
+/// Creating an [Equality] object that uses `employmentId`:
+///     new EqualityBy<Employee, int>((e) => e.employmentId);
```

Comment:
Maybe write this as `new EqualityBy((Employee e) => e.employmentId)`? Normally we don't type-annotate function parameters, but it's terser than explicitly annotating both type parameters.